elastic/beats
# [agentbeat] Fix agentbeat capabilities

## Proposed commit message

- Allow `agentbeat` to elevate permissions to match the permitted set.
- These are required for non-root use to operate agent in healthy state. 

## Checklist




## Disruptive User Impact




## How to test this PR locally
- https://github.com/elastic/elastic-agent/blob/main/docs/local-k8s-testing.md provides a good guide to test k8s locally.
- Clone the changes from this branch.
- Clone the changes from https://github.com/elastic/elastic-agent/pull/5271.
- Build a local docker image from `elastic-agent`.
- Install [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) and create a cluster
- Run `elastic-agent-standalone` from `elastic-agent` repo.
- Make sure to use `runAsUser: 1000` and add required capabilities in k8s manifest (SYS_PTRACE and DAC_OVERRIDE) like following:
```yaml
securityContext:
  runAsUser: 1000
  privileged: false
  capabilities:
   add:
    - DAC_OVERRIDE
    - SYS_PTRACE
    - SETPCAP
    - CHOWN
   drop:
    - ALL
```

## Related issues


- Relates https://github.com/elastic/elastic-agent/issues/5269


## Screenshots



Current behaviour, without enough capabilities:

<img width="1021" alt="Screenshot 2024-08-09 at 2 18 30 AM" src="https://github.com/user-attachments/assets/3b9bd82f-cab2-40b9-93ed-8eeee9df5e68">


Behaviour after the changes and https://github.com/elastic/elastic-agent/pull/5271:

<img width="591" alt="Screenshot 2024-08-09 at 2 20 37 AM" src="https://github.com/user-attachments/assets/de964ab0-04a5-4393-b26f-845747d0cac2">
